### PR TITLE
SKALE-4689 add is contract check and test

### PR DESCRIPTION
**Pull request by payvint**

### CODE REVIEW

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by yavrsky — DISMISSED** [review]

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by sergiy-skalelabs — DISMISSED** [review]

**Review by yavrsky — DISMISSED** [review]

File: proxy/migrations/upgradeMainnet.ts [hunk 1]
```diff
@@ -217,7 +217,33 @@ async function main() {
...
+                ));
+                safeTransactions.push(encodeTransaction(
+                    0,
+                    messageProxyForMainnetAddress,
+                    0,
+                    messageProxyForMainnet.interface.encodeFunctionData("setNewMessageGasCost", [9000])
+                ));
+            } else {
+                console.log(chalk.red("MessageProxyForMainnet was not found!"));
+                console.log(chalk.red("Check your abi!!!"));
```
Comment (DimaStebaev) [inline proxy/migrations/upgradeMainnet.ts:42]: Exit here.

File: proxy/test/extensions/ERC721MintingFromSchainToMainnet.ts [hunk 2]
```diff
@@ -369,6 +369,53 @@ describe("ERC721MintingFromSchainToMainnet", () => {
...
+        const resPost = await (await messageProxyForMainnet.connect(deployer).postIncomingMessages(
+            schainName,
+            0,
+            [message],
+            sign
+        )).wait();
+        if (!resPost.events) {
+            assert("No events were emitted");
+        } else {
+            expect(resPost.events[0]?.topics[0]).to.equal(stringValue(web3.utils.soliditySha3("PostMessageError(uint256,bytes)")));
```
Comment (DimaStebaev) [inline proxy/test/extensions/ERC721MintingFromSchainToMainnet.ts]: Do not hardcode event indexes.
Use `waffle` instead.
https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#emitting-events

File: proxy/test/extensions/ERC721MintingFromSchainToMainnet.ts [hunk 3]
```diff
@@ -369,6 +369,53 @@ describe("ERC721MintingFromSchainToMainnet", () => {
...
+        const dataToPost = await extensionSchain.connect(user).encodeParams(user.address, 1, "MyToken1");
+        const message = {
+            data: dataToPost,
+            destinationContract: user.address,
+            sender: extensionSchain.address,
+        };
+
+        // prepare BLS signature
+        // P.s. this is test signature from test of SkaleManager.SkaleVerifier - please do not use it!!!
+        const BlsSignature: [BigNumber, BigNumber] = [
```
Comment (DimaStebaev) [inline proxy/test/extensions/ERC721MintingFromSchainToMainnet.ts:384]: Use something obviously incorrect. For example (0, 0, 1, 0).

File: proxy/migrations/upgradeMainnet.ts [hunk 4]
```diff
@@ -217,7 +217,33 @@ async function main() {
...
+            const proxyAdmin = await getManifestAdmin(hre);
+            const owner = await proxyAdmin.owner();
+            const messageProxyForMainnetName = "MessageProxyForMainnet";
+            const messageProxyForMainnetFactory = await ethers.getContractFactory(messageProxyForMainnetName);
+            const messageProxyForMainnetAddress = abi[getContractKeyInAbiFile(messageProxyForMainnetName) + "_address"];
+            if (messageProxyForMainnetAddress) {
+                console.log(chalk.yellow("Prepare transaction to set message gas cost to 9000"));
+                const messageProxyForMainnet = messageProxyForMainnetFactory.attach(messageProxyForMainnetAddress) as MessageProxyForMainnet;
+                const constantSetterRole = await messageProxyForMainnet.CONSTANT_SETTER_ROLE();
+                safeTransactions.push(encodeTransaction(
```
Comment (DimaStebaev) [inline proxy/migrations/upgradeMainnet.ts]: Grant role only if it is needed.

**Review by DimaStebaev — CHANGES_REQUESTED** [review]

File: proxy/test/extensions/ERC721MintingFromSchainToMainnet.ts (hunk 3 above)
Comment (payvint) [reply proxy/test/extensions/ERC721MintingFromSchainToMainnet.ts:384]: Nope

**Review by yavrsky — APPROVED** [review]

**Review by DimaStebaev — APPROVED** [review]

**Pull request merged**